onnx_tf/backends/backend_v1.py

```diff
@@ -581,66 +583,73 @@ def handle_l_r_n(cls, node, input_dict):
-      if activations[0] != "sigmoid" or activations[1] != "tanh":
-        warnings.warn(
-            "Tensorflow uses sigmiod and tanh as first two activation functions."
-            "So activations attr will be set to sigmiod, tanh and {}.".format(
-                activations[2]))
-      tf_activations = [ONNX_OP_TO_TF_OP[activations[2]]]
+    else:
       if direction == "bidirectional":
-        if activations[3] != "sigmoid" or activations[4] != "tanh":
-          warnings.warn(
-              "Tensorflow uses sigmiod and tanh as first two activation functions."
-              "So activations attr will be set to sigmiod, tanh and {}.".format(
-                  activations[4]))
-        tf_activations.append(ONNX_OP_TO_TF_OP[activations[5]])
-
-    cell_kwargs["activation"] = tf_activations[0]
-    lstm_cell = tf.contrib.rnn.LSTMCell(hidden_size, **cell_kwargs)
-    cell = tf.contrib.rnn.MultiRNNCell([lstm_cell])
-    if direction == "bidirectional":
-      cell_kwargs["activation"] = tf_activations[1]
-      lstm_cell_bw = [tf.contrib.rnn.LSTMCell(hidden_size, **cell_kwargs)]
-      cell_bw = tf.contrib.rnn.MultiRNNCell([lstm_cell_bw])
-
-    # TODO: handle data types
-    if direction == "forward":
-      output, state = tf.nn.dynamic_rnn(
-          cell, input_dict[node.inputs[0]], time_major=True, dtype=tf.float32)
-    elif direction == "bidirectional":
-      output, state = tf.nn.bidirectional_dynamic_rnn(
-          cell,
-          cell_bw,
-          input_dict[node.inputs[0]],
-          time_major=True,
-          dtype=tf.float32)
-    elif direction == "reverse":
-
-      def _reverse(input_, seq_dim):
-        return array_ops.reverse(input_, axis=[seq_dim])
-
-      time_dim = 0
-      inputs_reverse = _reverse(input_dict[node.inputs[0]], time_dim)
-      output, state = tf.nn.dynamic_rnn(
-          cell, inputs_reverse, time_major=True, dtype=tf.float32)
-
-    state = state[0]
-    c, h = state
-    states = [h, c]
-    outputs = [output]
-    outputs.extend(states)
-    return outputs
+        activations = ["sigmoid", "tanh", "tanh", "sigmoid", "tanh", "tanh"]
+      else:
+        activations = ["sigmoid", "tanh", "tanh"]
+    clip = float(node.attrs["clip"]) if "clip" in node.attrs else None
+    hidden_size = int(node.attrs["hidden_size"])
+    input_forget = int(node.attrs.get("input_forget", "0"))
+    output_sequence = int(node.attrs.get("output_sequence", "0"))
+
+
+    # Read weights and create TensorFlow LSTMs, differentiating
+    # between the unidirectional and the bidirectional case
+    dummy_session = tf.Session()
+    if direction == "forward" or direction == "reverse":
+      W_weights = input_dict[W].eval(session=dummy_session)
+      R_weights = input_dict[R].eval(session=dummy_session)
+      B_weights = input_dict[B].eval(session=dummy_session)
+      tf_kernel, tf_bias = extract_onnx_lstm_weights(W_weights, R_weights,  B_weights)
```

Comment:
I have noticed another problem with my code, and that is the fact that it implies ranks of the LSTM input tensors (W, B, R) which do not match the ONNX specification. In my code, the tensors have rank 2 (`[4*hidden_size, input_size]`), whereas in the spec, they have rank 3 (`[num_directions, 4*hidden_size, input_size]`).

This problem stems from the fact that I have tested my code using the PyTorch exporter, which seems to generate tensors with incorrect ranks (`num_directions` is ignored). I will check back with the PyTorch team and update my code accordingly. Please do not merge the PR until then.